src/system/sys_profiler.cc

```diff
@@ -36,11 +36,23 @@
 #include "sys_profiler.hh"
 #include "system/system.hh"
 #include <boost/format.hpp>
+#include <iostream>
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/json_parser.hpp>
 
 #include "system/file_path.hh"
+#include "mpi.h"
+#include "time_point.hh"
 
 
-#include "mpi.h"
+using boost::property_tree::ptree;
+using boost::property_tree::read_json;
+using boost::property_tree::write_json;
+
+#define FLOW123D_JSON_PRETTY        1
```

Comment:
Why these defines.  Nobody knows their meaning without documentation. But better to use typed constants in particular functions, where the meaning could be clear from context. So two morals:
1) prefer const ints over defines
2) make global only what have to be global